src/components/ImportDialog.jsx

```diff
@@ -37,9 +38,30 @@ class ImportDialog extends Component {
 }
 
 ImportDialog.propTypes = {
-  onSubmit: PropTypes.func.isRequired,
   show: PropTypes.bool.isRequired,
-  onCancel: PropTypes.func.isRequired,
+  onCloseButtonClick: PropTypes.func.isRequired,
+  onImportButtonClick: PropTypes.func.isRequired,
+  onImportableChordSheetChange: PropTypes.func.isRequired,
+};
+
+const mapStateToProps = (state) => {
+  const { showImportDialog: show } = state.ui;
+  return { show };
 };
 
-export default ImportDialog;
+const mapDispatchToProps = dispatch => ({
+  onCloseButtonClick() {
+    dispatch(hideImportDialog());
+  },
+
+  onImportableChordSheetChange(event) {
+    dispatch(setImportableChordSheet(event.target.value));
+  },
+
+  onImportButtonClick() {
+    dispatch(importChordSheet());
```

Comment:
This should also be moved, it's view responsbility when this happens. If you want to be a perfectionist. you can use middleware for this